server/tests/organization/test_service.py

```diff
@@ -163,19 +163,126 @@ async def test_valid_with_none_subscription_settings(
 
 
 @pytest.mark.asyncio
-async def test_get_next_invoice_number(
+async def test_get_next_invoice_number_organization(
     session: AsyncSession,
     organization: Organization,
+    customer: Customer,
 ) -> None:
     assert organization.customer_invoice_next_number == 1
 
     next_invoice_number = await organization_service.get_next_invoice_number(
-        session, organization
+        session, organization, customer
     )
 
     assert next_invoice_number == f"{organization.customer_invoice_prefix}-0001"
     assert organization.customer_invoice_next_number == 2
 
+    await session.refresh(customer)
+    assert customer.invoice_next_number == 1
+
+
+@pytest.mark.asyncio
+async def test_get_next_invoice_number_customer(
+    session: AsyncSession,
+    save_fixture: SaveFixture,
+    organization: Organization,
+    customer: Customer,
+) -> None:
+    organization.subscription_settings = {
+        **organization.subscription_settings,
+        "invoice_numbering": InvoiceNumbering.customer,
+    }
+    await save_fixture(organization)
+
+    initial_org_counter = organization.customer_invoice_next_number
+    assert customer.invoice_next_number == 1
+
+    next_invoice_number = await organization_service.get_next_invoice_number(
+        session, organization, customer
+    )
+
+    assert (
+        next_invoice_number
+        == f"{organization.customer_invoice_prefix}-{customer.short_id_str}-0001"
+    )
+    await session.commit()
```

Comment:
Not sure it's necessary to commit/refresh everywhere.

Besides, I would _highly_ recommend to prefer `session.flush()` over `session.commit()` in most cases :D